digitalocean/resource_digitalocean_floating_ip.go

```diff
@@ -38,6 +38,11 @@ func resourceDigitalOceanFloatingIp() *schema.Resource {
 				Type:     schema.TypeInt,
 				Optional: true,
 			},
+
+			"ignore_droplet_id": {
```

Comment:
This feels like it may be a bit overly complicated. Currently if the `droplet_id` attribute is not set, the resource does not attach the IP to a Droplet. I think it would be more straightforward to document that in order to assign a Floating IP to a Droplet you should use either the `droplet_id` attribute in the `floating_ip` resource or the `floating_ip_assignment` but not both. Same as in the `volume_attachment` and `droplet` resource.

That would match the approach to the `volume_attachment` resource that recently landed. See the notes here:

https://github.com/terraform-providers/terraform-provider-digitalocean/blob/master/website/docs/r/volume_attachment.html.markdown#digitalocean_volume_attachment
https://github.com/terraform-providers/terraform-provider-digitalocean/blob/master/website/docs/r/droplet.html.markdown#argument-reference